packages/waas/src/auth.ts

```diff
@@ -221,24 +223,71 @@ export class SequenceWaaS {
     return this.waas.isSignedIn()
   }
 
-  async signIn(creds: Identity, name: string): Promise<SignInResponse> {
-    // TODO: Be smarter about this, for cognito (or some other cases) we may
-    // want to send the email instead of the idToken
-    const signInIntent = await this.waas.signIn({
-      idToken: creds.idToken
-    })
+  async signIn(creds: IdTokenIdentity | string, sessionName: string): Promise<SignInResponse> {
+    const idToken = (typeof creds === 'string' ? creds : creds.idToken)
 
-    // Login on WaaS
-    const decoded = jwtDecode(creds.idToken)
+    const intent = await this.waas.signInWithIdToken(idToken)
+    try {
+      const res = await this.registerSession(intent, sessionName)
+
+      await this.waas.completeSignIn({
```

Comment:
is `completeSignIn` like a callback..? or is it for responding with the challenge..?

Reply:
It's an internal method that updates the state to "authenticated" and stores the result of the `registerSession` call. The flow is actually:

1. `initiateAuth` - commit to the login and receive a challenge (e.g. for email auth the code is sent at this point)
2. `registerSession` - respond to the challenge (e.g. for email auth, send the answer)

This `signIn` method is used for legacy OIDC auth that skips the first part of the flow and moves on to `registerSession` immediately.